app/response_processor.py

```diff
@@ -80,6 +82,18 @@ def _send_receipt(self, decrypted, data):
                 res_logger.error("Receipt rejected by endpoint")
                 raise BadMessageError("Failure to send receipt")
 
+            elif res.status_code == 404:
+                namespaces = {'error': 'http://ns.ons.gov.uk/namespaces/resources/error'}
+                tree = etree.fromstring(res.content)
+                element = tree.find('error:message', namespaces).text
+
+                if '1009' in element:
+                    res_logger.error("Receipt rejected by endpoint", error=1009)
```

Comment:
Now logging three key:val elements: error, stat_unit_id and collection_exercise_sid. Also expanded on the event message.